src/middlewared/middlewared/plugins/jail.py

```diff
@@ -36,6 +36,469 @@
+        iocroot = IOCJson(pool).json_get_value('iocroot')
+        plugin_dir_path = os.path.join(iocroot, '.plugins')
+        plugin_jails = {
+            j['host_hostuuid']: j for j in self.middleware.call_sync(
+                'jail.query', [['type', 'in', ['plugin', 'pluginv2']]]
+            )
+        }
+
+        index_jsons = {}
+        for repo in os.listdir(plugin_dir_path) if os.path.exists(plugin_dir_path) else []:
+            index_path = os.path.join(plugin_dir_path, repo, 'INDEX')
+            if os.path.exists(index_path):
+                with contextlib.suppress(json.decoder.JSONDecodeError):
+                    with open(index_path, 'r') as f:
+                        index_jsons[repo] = json.loads(f.read())
+
+        for index, plugin in enumerate(resource_list):
+            # "plugin" is a list which we will convert to a dictionary for readability
+            plugin_dict = {
+                k: v if v != '-' else None
+                for k, v in zip((
+                    'jid', 'name', 'boot', 'state', 'type', 'release', 'ip4', 'ip6', 'template', 'admin_portal'
+                ), plugin)
+            }
+            plugin_output = pathlib.Path(f'{iocroot}/jails/{plugin_dict["name"]}/root/root/PLUGIN_INFO')
+            plugin_info = plugin_output.read_text().strip() if plugin_output.is_file() else None
+
+            plugin_name = plugin_jails[plugin_dict['name']]['plugin_name']
+            plugin_repo = self.convert_repository_to_path(plugin_jails[plugin_dict['name']]['plugin_repository'])
+            plugin_dict.update({
+                'id': plugin_dict['name'],
+                'plugin_info': plugin_info,
+                'plugin': plugin_name,
+                'plugin_repository': plugin_jails[plugin_dict['name']]['plugin_repository'],
+                **self.get_local_plugin_version(
+                    plugin_name,
+                    index_jsons.get(plugin_repo), iocroot, plugin_dict['name']
+                )
+            })
+
+            resource_list[index] = plugin_dict
+
+        return filter_list(resource_list, filters, options)
+
+    @accepts(
+        Dict(
+            'plugin_create',
+            Str('plugin_name', required=True),
+            Str('jail_name', required=True),
+            List('props', default=[], empty=False),
+            Str('branch', default=None, null=True),
+            Str('plugin_repository', default='https://github.com/freenas/iocage-ix-plugins.git'),
+        )
+    )
+    def do_create(self, data):
+        """
+        Create a Plugin.
+
+        `plugin_name` is the name of the plugin as specified by the INDEX file in "plugin_repository" and it's json
+        file.
+
+        `jail_name` is the name of the plugin jail and is required.
+
+        `props` is a list of iocage props which user can manually set. For plugins, it should be ensured that the
+        jails have networking capability set either with dhcp, ip address or nat.
```

Comment:
```suggestion
        jail networking capability with DHCP, IP Address, or NAT properties.
```
Might be helpful to have an example (dhcp=on, ip4_addr="em0|192.168.0.1/24", etc) or link to iocage manpage?